lib/connectionContext.ts

```diff
@@ -75,17 +86,54 @@ export namespace ConnectionContext {
    */
   export const userAgent: string = "/js-event-hubs";
 
-  export function create(config: ConnectionConfig, tokenProvider?: TokenProvider): ConnectionContext {
+  function encodeMessage(body: any): any {
+    let result: any;
+    if (body != undefined)
+      if (isBuffer(body))
+        result = rhea.message.data_section(body);
+      else if (typeof body !== "string")
+        result = rhea.message.data_section(Buffer.from(JSON.stringify(body), "utf8"));
+      else
+        result = rhea.message.data_section(Buffer.from(body, "utf8"));
+    else
+      result = rhea.message.data_section(Buffer.from("", "utf8"));
+
+    return result;
+  }
+
+  function decodeMessage(body: any): any {
+    let processedBody: any = body;
+    if (isBuffer(body)) {
+      processedBody = body.toString("utf8");
+    } else if (typeof body === "string") {
+      processedBody = body;
+    } else if (body.content) {
```

Comment:
what's body.content?

Reply:
It is how the body comes back when we encode it as a "data_section" in rhea terms. The body is a [`Section` object](https://github.com/amqp/rhea/blob/master/lib/message.js#L74) with properties `typecode, content and multiple`. Until now, I have found the content is usually a buffer. I have seen customers doing `EventData.body.content.toString("utf8")`